scripts/leapfrog/README.md

```diff
@@ -0,0 +1,136 @@
+The RPCO leapfrog scripts are a thin wrapper around OSA-OPS leapfrog tools. 
+
+For details please refer to the scripts themselves. Paths are omitted for
+brevity, scripts may not be in the root of the relevant repo.
+
+
+#### Pre Leap
+This step removes modifications to RPCO Kilo that aren't compatible with OSA
+Newton. Currently this only contains an Ansible 1.9 compatibility workaround.
+
+#### Prep
+This step executes pre-flight checks, and prompts the user for confirmation. It
+also ensures that the databases are backed up.
+
+#### Upgrade
+This step has a section for each major version between the source (Kilo) and
+target (Newton) versions. Each section includes:
+* Variable & Secrets Migration (OSA)
+* Fact Cleanup
+* Hostname Compatibility checks/modifications.
+
+#### Migrations
+The step runs the database migrations for each major upgrade in turn.
+For a RPC Kilo --> Newton upgrade this involves running the liberty,
+mitaka and newton migrations in sequence.
+
+#### Re-Deploy
+This step first runs the pre_re deploy script which handles RPC variable and
+secret migrations. Next the target version of OSA is deployed.
+
+#### Post Leap
+This step deploys the RPC additions on top of the already deployed OSA.
+
+## Rollback and Interruption
+The Leapfrog upgrade is a one way process, once it has begun there is no
+automated rollback. If a leapfrog upgrade is interrupted, it can be resumed.
+Each major step of the leapfrog upgrade process creates a marker file, which
+will be used to skip completed tasks on subsequent runs.
+
+If a step fails information about that step will be printed along with all the
+remaining steps. The operator must fix the failure before re-running the
+leapfrog or completing the remaining steps manually.
+
+## Confirmation Prompts
+The leapfrog process does not run unattended, it requires the operator to
+confirm at various points. This is to ensure that an irreversible leapfrog is
+not initiated accidentally.
+
+
+## Problems
+
+#### Repo Build
+The redeploy phase involves many git clones from github, this may cause rate
+limits, which in turn can cause timeouts and the repo build process to fail. If
+this happens, wait for rate limits to reset before retrying.
+
+#### Galera
+Occasionally the galera cluster may be in a non-functional state after the
+leapfrog. If this is the case, follow the [Galera Maintenance section of the OSA
+operations
+guide](https://docs.openstack.org/openstack-ansible/latest/admin/maintenance-tasks/galera.html).
+
+#### Confirmation Prompts and the Ctrl-c warning
+The confirmation prompts instruct the user not to interrupt the leapfrog process
+via ctrl-c. While an uninterrupted upgrade is the smoothest, the consequences
+of interruption are not as dire as implied. The process can be resumed by
+re-running the top level script, which will skip the steps that have already
+been completed by checking for the existence of marker files.
+
```

Comment:
Keep in mind, when resuming, that the detection system could say that you are running Newton, even if leapfrog the process didn't fully run.
That's normal, because some of the files used for detection could have been migrated (depending on when the Ctrl-C happened), while the rest haven't been migrated.

Please continue the process as if you were already on newton, to ensure everything ran, if something happened after the upgrade steps.